sdk/storage/storage-file-datalake/README.md

```diff
@@ -0,0 +1,520 @@
+This library depends on following ES features which need external polyfills loaded.
+
+- `Promise`
+- `String.prototype.startsWith`
+- `String.prototype.endsWith`
+- `String.prototype.repeat`
+- `String.prototype.includes`
+- `Array.prototype.includes`
+- `Object.assign`
+- `Object.keys` (Override IE11's `Object.keys` with ES6 polyfill forcely to enable [ES6 behavior](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Object/keys#Notes))
+- `Symbol`
+
+#### Differences between Node.js and browsers
+
+There are differences between Node.js and browsers runtime. When getting started with this library, pay attention to APIs or classes marked with _"ONLY AVAILABLE IN NODE.JS RUNTIME"_ or _"ONLY AVAILABLE IN BROWSERS"_.
+
+##### Features, interfaces, classes or functions only available in Node.js
+
+- Shared Key Authorization based on account name and account key
+  - `StorageSharedKeyCredential`
+- Shared Access Signature(SAS) generation
+  - `generateAccountSASQueryParameters()`
+  - `generateDataLakeSASQueryParameters()`
+
+##### Features, interfaces, classes or functions only available in browsers
+
+- N/A
+
+### JavaScript Bundle
+
+To use this client library in the browser, you need to use a bundler. For details on how to do this, please refer to our [bundling documentation](https://aka.ms/AzureSDKBundling).
+
+### CORS
+
+You need to set up [Cross-Origin Resource Sharing (CORS)](https://docs.microsoft.com/rest/api/storageservices/cross-origin-resource-sharing--cors--support-for-the-azure-storage-services) rules for your storage account if you need to develop for browsers. Go to Azure portal and Azure Storage Explorer, find your storage account, create new CORS rules for blob/queue/file/table service(s).
+
+For example, you can create following CORS settings for debugging. But please customize the settings carefully according to your requirements in production environment.
+
+- Allowed origins: \*
+- Allowed verbs: DELETE,GET,HEAD,MERGE,POST,OPTIONS,PUT
+- Allowed headers: \*
+- Exposed headers: \*
+- Maximum age (seconds): 86400
+
+> Notice: Data Lake currently shares CORS settings for blob service.
+
+## Examples
+
+### Import the package
+
+To use the clients, import the package into your file:
+
+```javascript
+const AzureStorageDataLake = require("@azure/storage-file-datalake");
+```
+
+Alternatively, selectively import only the types you need:
+
+```javascript
+const { DataLakeServiceClient, StorageSharedKeyCredential } = require("@azure/storage-file-datalake");
+```
+
+### Create the data lake service client
+
+Use the constructor to create a instance of `DataLakeServiceClient`.
+
+- **Recommended way to instantiate a `DataLakeServiceClient` - with `DefaultAzureCredential` from `@azure/identity` package**
+
+  Setup : Reference - Authorize access to blobs and queues with Azure Active Directory from a client application - https://docs.microsoft.com/azure/storage/common/storage-auth-aad-app
```

Comment:
adls gen2 should has it's own page for ACL and OAuth usage.

Reply:
Actually, these docs are appliable for adls, it reuse blob related roles, see

![image](https://user-images.githubusercontent.com/6889160/70196878-01e21f00-1745-11ea-8382-7d87dcd9d1a8.png)

Reply:
Added a notice about datalake reuses blob permissions during OAuth.